hw3/bleichenbacher.py

```diff
@@ -134,12 +169,15 @@ def narrow_m(key, m_prev, s, B):
     """
     intervals = []
     for a, b in m_prev:
-        min_r = ?
-        max_r = ?
+        min_r = divceil((a * s) - (3 * B) + 1, key.n)
+        max_r = divfloor((b * s) - (2 * B), key.n)
         for r in range(min_r, max_r + 1):
-            start = ?
-            end = ?
-            intervals.append((start, end))
+            start = max(a, divceil((2 * B) + (r * key.n), s))
+            end = min(b, divfloor((3 * b) - 1 + (r * key.n), s))

```

Comment:
```suggestion
            end = min(b, divfloor((3 * B) - 1 + (r * key.n), s))
```

Really weird that it worked.. but I guess B and b are around the same size pretty fast so it would fail only if we're unlucky?